fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java

```diff
@@ -180,6 +180,14 @@ public void initialize(BDBEnvironment environment, String baseImageDir) throws I
 
         // load meta
         loadImage(imageDir);
+
+        // Fully set up the starOsAgent so that we don't need to call `StarOSAgent.prepare` elsewhere
+        LOG.info("Waiting for starMgrServer to start");
+        starMgrServer.blockUntilStart();
+        LOG.info("Done waiting for starMgrServer to start");
+        if (starOsAgent != null) {
+            starOsAgent.prepare();
```

Comment:
this won't work in case of a refresh installed shared-data cluster, the follower will not be able to success the prepare() op until it has fully synced with the leader where the leader has done the service registration and bootstrap.

Reply:
Thanks for catching this -- I moved the call to after GlobalStateMgr is ready, which I believe includes waiting for metadata sync with the leader. Please let me know if it's possible that the leader still hasn't registered the service with StarMgr at this point.